xarray/core/dataset.py

```diff
@@ -2337,21 +2337,36 @@ def expand_dims(self, dim, axis=None):
 
         Parameters
         ----------
-        dim : str or sequence of str.
+        dim : str, sequence of str, dict, None
             Dimensions to include on the new variable.
-            dimensions are inserted with length 1.
+            If provided as str or sequence of str, then dimensions are inserted
+            with length 1. If provided as a dict, then the keys are the new
+            dimensions and the values are either integers (giving the length of
+            the new dimensions) or iterables (giving the coordinates of the new
+            dimensions).
         axis : integer, list (or tuple) of integers, or None
             Axis position(s) where new axis is to be inserted (position(s) on
             the result array). If a list (or tuple) of integers is passed,
             multiple axes are inserted. In this case, dim arguments should be
             the same length list. If axis=None is passed, all the axes will
             be inserted to the start of the result array.
+        **new_dims : int or  iterable
```

Comment:
Maybe `int or sequence/ndarray`?

Iterables actually are not valid inputs into `np.array()`.